includes/options/class-amp-options-menu.php

```diff
@@ -272,45 +281,168 @@ public function render_validation_handling() {
 	}
 
 	/**
-	 * Post types support section renderer.
+	 * Supported templates section renderer.
 	 *
-	 * @since 0.6
+	 * @since 1.0
 	 */
-	public function render_post_types_support() {
-		$builtin_support = AMP_Post_Type_Support::get_builtin_supported_post_types();
-		$element_name    = AMP_Options_Manager::OPTION_NAME . '[supported_post_types][]';
+	public function render_supported_templates() {
+		$theme_support_args = AMP_Theme_Support::get_theme_support_args( array( 'initial' => true ) ); // Initial so we can get before removed if optional.
+
 		?>
+		<fieldset id="all_templates_supported_fieldset">
+			<?php if ( isset( $theme_support_args['templates_supported'] ) && 'all' === $theme_support_args['templates_supported'] ) : ?>
+				<div class="notice notice-info notice-alt inline">
+					<p>
+						<?php esc_html_e( 'The current theme requires all templates to support AMP.', 'amp' ); ?>
+					</p>
+				</div>
+			<?php else : ?>
+				<p>
+					<label for="all_templates_supported">
+						<input id="all_templates_supported" type="checkbox" name="<?php echo esc_attr( AMP_Options_Manager::OPTION_NAME . '[all_templates_supported]' ); ?>" <?php checked( AMP_Options_Manager::get_option( 'all_templates_supported' ) ); ?>>
+						<?php esc_html_e( 'Serve all templates as AMP regardless of what is being queried.', 'amp' ); ?>
+					</label>
+				</p>
+				<p class="description">
+					<?php esc_html_e( 'This will allow all of the URLs on your site to be served as AMP by default.', 'amp' ); ?>
+				</p>
+			<?php endif; ?>
+		</fieldset>
+
+		<fieldset id="supported_post_types_fieldset">
+			<?php $element_name = AMP_Options_Manager::OPTION_NAME . '[supported_post_types][]'; ?>
+			<h4 class="title"><?php esc_html_e( 'Content Types', 'amp' ); ?></h4>
+			<p>
+				<?php esc_html_e( 'The following content types will be available as AMP:', 'amp' ); ?>
+			</p>
+			<ul>
+			<?php foreach ( array_map( 'get_post_type_object', AMP_Post_Type_Support::get_eligible_post_types() ) as $post_type ) : ?>
+				<li>
+					<?php $element_id = AMP_Options_Manager::OPTION_NAME . "-supported_post_types-{$post_type->name}"; ?>
```

Comment:
**Media Pages In "Classic" Mode**

This isn't the most common use-case. But it looks like on the "Media" pages in "Classic" mode, the `href` in the `<link rel="amphtml">` leads to a 404 page. It ends with `/amp`, which results in a 404 page. But the `?amp` URL works.

Steps to reproduce:
1. Activate Twenty Sixteen
2. On the "AMP Settings" page, select "Template Mode" > "Classic," and check "Supported Templates" > "Media"
<img width="917" alt="classic-media-templates" src="https://user-images.githubusercontent.com/4063887/42179630-49d82556-7dfa-11e8-890b-b9b74a3248c9.png">
3. In the Media Library, click an image, and click "View Attachment Page":
<img width="1395" alt="media-page" src="https://user-images.githubusercontent.com/4063887/42179697-82d0e244-7dfa-11e8-9860-b7f671dc5dcb.png">

4. In the source of that page, the `<link rel="amphtml">` has an `href` with `/amp`, not `?amp`:

```html
<link rel="amphtml" href="https://loc.test/lorem-2000bridge/amp/">
```
<img width="797" alt="source-media-page" src="https://user-images.githubusercontent.com/4063887/42179898-d6eb2088-7dfa-11e8-9cbe-8bbb4526f8dc.png">

5. Expected: that URL is a valid AMP page
6. Actual: it's a 404 page:
<img width="962" alt="ends-amp" src="https://user-images.githubusercontent.com/4063887/42180164-ca47987e-7dfb-11e8-9a7d-b0f5b344eee9.png">

7. Going to the same URL, but ending in `?amp` results in a valid AMP page:
<img width="878" alt="ends-amp-query-var" src="https://user-images.githubusercontent.com/4063887/42180220-083eb194-7dfc-11e8-9261-dc9dc3a749fe.png">